apps/meteor/app/api/server/v1/chat.ts

```diff
@@ -282,6 +282,71 @@ const chatEndpoints = API.v1
 
 			return API.v1.success();
 		},
+	)
+	.post(
+		'chat.update',
+		{
+			authRequired: true,
+			body: isChatUpdateProps,
+			response: {
+				400: validateBadRequestErrorResponse,
+				401: validateUnauthorizedErrorResponse,
+				200: ajv.compile<{ message: IMessage }>({
+					type: 'object',
+					properties: {
+						message: { $ref: '#/components/schemas/IMessage' },
+						success: {
+							type: 'boolean',
+							enum: [true],
+						},
+					},
+					required: ['message', 'success'],
+					additionalProperties: false,
+				}),
+			},
+		},
+		async function action() {
+			const { bodyParams } = this;
+
+			const msg = await Messages.findOneById(bodyParams.msgId);
+
+			// Ensure the message exists
+			if (!msg) {
+				return API.v1.failure(`No message found with the id of "${bodyParams.msgId}".`);
+			}
+
+			if (bodyParams.roomId !== msg.rid) {
+				return API.v1.failure('The room id provided does not match where the message is from.');
+			}
+
+			const updateData: Parameters<typeof executeUpdateMessage> = [
+				this.userId,
+				'content' in bodyParams
```

Comment:
Maybe instead of checking for `content` in `bodyParams`, we should check if the `msg` from database is encrypted and handle the case accordingly.

Because if we do this, then the API can be misused by sending `contents` param for a non-encrypted message.